pass-nihms-loader/nihms-pass-client/src/main/java/org/eclipse/pass/client/nihms/NihmsPassClientService.java

```diff
@@ -0,0 +1,659 @@
+/*
+ * Copyright 2023 Johns Hopkins University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.pass.client.nihms;
+
+import static org.eclipse.pass.loader.nihms.util.ProcessingUtil.nullOrEmpty;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.pass.client.nihms.cache.GrantIdCache;
+import org.eclipse.pass.client.nihms.cache.NihmsDepositIdCache;
+import org.eclipse.pass.client.nihms.cache.NihmsRepositoryCopyIdCache;
+import org.eclipse.pass.client.nihms.cache.PublicationIdCache;
+import org.eclipse.pass.client.nihms.cache.UserPubSubmissionsCache;
+import org.eclipse.pass.loader.nihms.util.ConfigUtil;
+import org.eclipse.pass.support.client.PassClient;
+import org.eclipse.pass.support.client.PassClientResult;
+import org.eclipse.pass.support.client.PassClientSelector;
+import org.eclipse.pass.support.client.RSQL;
+import org.eclipse.pass.support.client.model.Deposit;
+import org.eclipse.pass.support.client.model.Grant;
+import org.eclipse.pass.support.client.model.Journal;
+import org.eclipse.pass.support.client.model.Publication;
+import org.eclipse.pass.support.client.model.Repository;
+import org.eclipse.pass.support.client.model.RepositoryCopy;
+import org.eclipse.pass.support.client.model.Submission;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * NIHMS PASS client service deals with interactions with the data via the PASS client and controls local data caches
+ *
+ * @author Karen Hanson
+ */
+public class NihmsPassClientService {
```

Comment:
This class is pretty long.  Consider refactoring into maybe a read and write service.

Reply:
Yes, I agree, its a bit lengthy. I can address this now or add it to the clean up?

Reply:
It's up to you, if it would be quick (couple hours), may be worth doing now.